ddl/partition.go

```diff
@@ -112,6 +112,26 @@ func checkPartitionNameUnique(tbInfo *model.TableInfo, pi *model.PartitionInfo)
 	return nil
 }
 
+func checkPartitionFuncValid(expr ast.ExprNode) error {
```

Comment:
The expression should return integer value:

> Data type of partitioning key.  A partitioning key must be either an integer column or an expression that resolves to an integer. Expressions employing ENUM columns cannot be used. The column or expression value may also be NULL. (See Section 3.7, “How MySQL Partitioning Handles NULL”.)
https://dev.mysql.com/doc/mysql-partitioning-excerpt/5.7/en/partitioning-limitations.html

Reply:
done